crates/sui-framework/sources/priority_queue.move

```diff
@@ -105,6 +125,16 @@ module sui::priority_queue {
         pragma opaque;
     }
 
+    public fun priorities<T: drop>(pq: &PriorityQueue<T>): vector<u64> {
+        let res = vector[];
+        let i = 0;
+        while (i < vector::length(&pq.entries)) {
+            vector::push_back(&mut res, vector::borrow(&pq.entries, i).priority);
+            i = i +1;
+        };
+        res
+    }
+
     #[test]
     fun test_pq() {
```

Comment:
Can we add a test that would have failed in the previous set-up and passes now?

Reply:
yes working on it

Reply:
added a test case with the real data that failed the previous setup